CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+  if (DEFINED BOOST_ROOT)
+    if (CMAKE_SIZEOF_VOID_P EQUAL 8 AND IS_DIRECTORY ${BOOST_ROOT}/stage64/lib)
+      set (BOOST_LIBRARYDIR ${BOOST_ROOT}/stage64/lib)
+    else ()
+      set (BOOST_LIBRARYDIR ${BOOST_ROOT}/stage/lib)
+    endif ()
+  endif ()
+endif ()
+message (STATUS "BOOST_ROOT: ${BOOST_ROOT}")
+message (STATUS "BOOST_LIBRARYDIR: ${BOOST_LIBRARYDIR}")
+
+# uncomment the following as needed to debug FindBoost issues:
+#set (Boost_DEBUG ON)
+
+#[=========================================================[
+   boost dynamic libraries don't trivially support @rpath
+   linking right now (cmake's default), so just force
+   static linking for macos, or if requested on linux by flag
+#]=========================================================]
+if (static OR APPLE OR MSVC)
+  set (Boost_USE_STATIC_LIBS ON)
+endif ()
+set (Boost_USE_MULTITHREADED ON)
+if (static OR MSVC)
+  set (Boost_USE_STATIC_RUNTIME ON)
+else ()
+  set (Boost_USE_STATIC_RUNTIME OFF)
+endif ()
+find_package (Boost 1.67 REQUIRED
+  COMPONENTS
+    chrono
+    context
+    coroutine
+    date_time
+    filesystem
+    program_options
+    regex
+    serialization
+    system
+    thread)
+
+add_library (ripple_boost INTERFACE)
+add_library (Ripple::boost ALIAS ripple_boost)
+if (is_xcode)
+  target_include_directories (ripple_boost BEFORE INTERFACE ${Boost_INCLUDE_DIRS})
+  target_compile_options (ripple_boost INTERFACE --system-header-prefix="boost/")
+else ()
+  target_include_directories (ripple_boost SYSTEM INTERFACE ${Boost_INCLUDE_DIRS})
 endif()
 
-add_with_props(rippled_src_all src/ripple/unity/secp256k1.cpp
-  -I"${CMAKE_SOURCE_DIR}/"src/secp256k1
-  ${no_unused_w}
-  )
-
-foreach(cursrc
-    src/ripple/unity/beast.cpp
-    src/ripple/unity/lz4.c
-    src/ripple/unity/protobuf.cpp
-    src/ripple/unity/ripple.proto.cpp)
-
-  add_with_props(rippled_src_all ${cursrc}
-    ${rocks_db_system_header}
+target_link_libraries (ripple_boost
+  INTERFACE
+    Boost::boost
+    Boost::chrono
+    Boost::coroutine
+    Boost::date_time
+    Boost::filesystem
+    Boost::program_options
+    Boost::regex
+    Boost::serialization
+    Boost::system
+    Boost::thread)
+
+#[===================================================================[
+   NIH dep: openssl
+#]===================================================================]
+
+if (APPLE AND NOT DEFINED ENV{OPENSSL_ROOT_DIR})
+  find_program (HOMEBREW brew)
+  if (NOT HOMEBREW STREQUAL "HOMEBREW-NOTFOUND")
+    execute_process (COMMAND ${HOMEBREW} --prefix openssl
+      OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+  endif ()
+endif ()
+
+if ((NOT DEFINED OPENSSL_ROOT) AND (DEFINED ENV{OPENSSL_ROOT}))
+  set (OPENSSL_ROOT $ENV{OPENSSL_ROOT})
+endif ()
+file (TO_CMAKE_PATH "${OPENSSL_ROOT}" OPENSSL_ROOT)
+
+if (static OR APPLE OR MSVC)
+  set (OPENSSL_USE_STATIC_LIBS ON)
+endif ()
+set (OPENSSL_MSVC_STATIC_RT ON)
+set (_ssl_min_ver 1.0.2)
+# HACK for travis
```

Comment:
Why can't we just update the package version of openssl on Travis?

Reply:
https://docs.travis-ci.com/user/reference/overview/ - Travis still only offers Precise and Trusty.

Trusty only offers OpenSSL 1.0.1f (https://packages.ubuntu.com/source/trusty/openssl)

Travis seems unwilling/unable to update their base system (https://github.com/travis-ci/travis-ci/issues/5821), a workaround would be to run docker inside a VM instance and build/test in there.

Alternatively you could download/install a deb package from a newer Ubuntu version and hope it works.